includes/templates/class-amp-post-template.php

```diff
@@ -302,15 +302,19 @@ private function build_post_data() {
 	 * @return int Post publish UTC timestamp.
 	 */
 	private function build_post_publish_timestamp() {
-		$format    = 'U';
-		$timestamp = (int) get_post_time( $format, true, $this->post, true );
+		if ( empty( $this->post->post_date_gmt ) || '0000-00-00 00:00:00' === $this->post->post_date_gmt ) {
```

Comment:
@westonruter Do you think a check should be added to only use `time()` when the post status is either `draft`, `pending` or `auto-draft`?

Reply:
I don't think that is necessary, as that special value would only ever happen for a non-published post anyway. 

Apparently this post statuses that include that special value are called `date_floating`, which means “whether the post has a floating creation date.” If it has no creation date, then `now()` should be assumed no?

https://github.com/WordPress/wordpress-develop/blob/19513ff6a0fc27b2ff7e9debeb6e94b7e1b4d658/src/wp-includes/post.php#L3775-L3810

Reply:
Oh interesting. So yes, if no creation date, use `time()` (assuming that you mistakenly used `now()` for `time()`).

Reply:
Yes, I meant `time()`.